Plugins/Public/base_plugin/PlayerCommands.cpp

```diff
@@ -1825,6 +1828,52 @@ namespace PlayerCommands
 		}
 	}
 
+	void GetNecessitiesStatus(uint client, const wstring &args) {
+		// Check that this player is in a player controlled base
+		PlayerBase *base = GetPlayerBaseForClient(client);
+		if (!base)
+		{
+			PrintUserCmdText(client, L"ERR Not in player base");
+			return;
+		}
+
+		if (!clients[client].admin && !clients[client].viewshop)
+		{
+			PrintUserCmdText(client, L"ERROR: Access denied");
+			return;
+		}
+
+		if (base->HasMarketItem(set_base_crew_type) < base->base_level * 200) {

```

Comment:
I would be slightly happier with:
```
uint baseItemCount = base->HasMarketItem(set_base_crew_type);
uint baseItemNeed = base->base_level * 200;
if (baseItemCount < baseItemNeed) {
```
That way we would avoid multiple operations like this (relying on the compiler to get the order of operations right instead of being unambiguous, though I suppose with these types it couldn't work any other way), and we could also reuse baseItemCount below too.

Reply:
Agreed, submitted.